framework/base/src/types/interaction/tx_exec/tx_exec_upgrade.rs

```diff
@@ -133,3 +168,58 @@ where
         });
     }
 }
+
+impl<Api, Payment, Gas, RH>
```

Comment:
Same here. Generalize using `TxToSpecified`, not by duplicating the entire call.